src/java.net.http/share/classes/jdk/internal/net/http/HttpResponseImpl.java

```diff
@@ -44,6 +44,7 @@
 class HttpResponseImpl<T> implements HttpResponse<T>, RawChannel.Provider {
 
     final int responseCode;
+    private final Optional<String> connectionLabel;
```

Comment:
There's a precedent for `previousResponse` below, but we could fix that in a different cleanup PR, and avoid introducing new fields of type `Optional`.

```suggestion
    private final String connectionLabel;
```